src/order/succ_pred.lean

```diff
@@ -0,0 +1,143 @@
+/-
+Copyright (c) 2021 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import order.bounded_lattice
+
+/-!
+# Successor and predecessor
+
+-/
+
+/-- Order equipped with a sensible successor function. -/
+@[ext] class succ_order (α : Type*) [preorder α] :=
+(succ : α → α)
+(le_succ : ∀ a, a ≤ succ a)
+(maximal_of_succ_le : ∀ ⦃a⦄, succ a ≤ a → ∀ ⦃b⦄, ¬a < b)
+(succ_le_of_lt : ∀ {a b}, a < b → succ a ≤ b)
+(le_of_lt_succ : ∀ {a b}, a < succ b → a ≤ b)
+
+open succ_order
+
+variables {α : Type*}
+
+/-- A constructor taking -/
+def succ_order_of_lt_succ_iff_le_of_lt_iff_succ_le [preorder α] {succ : α → α}
+  (hlt_iff_succ_le : ∀ {a b}, a < b ↔ succ a ≤ b)
+  (hlt_succ_iff_le : ∀ {a b}, a < succ b ↔ a ≤ b) :
+  succ_order α :=
+{ succ := succ,
+  le_succ := λ a, (hlt_succ_iff_le.2 le_rfl).le,
+  maximal_of_succ_le := λ a ha, (lt_irrefl a (hlt_iff_succ_le.2 ha)).elim,
+  succ_le_of_lt := λ a b, hlt_iff_succ_le.1,
+  le_of_lt_succ := λ a b, hlt_succ_iff_le.1 }
+
+section preorder
+variables [preorder α] [succ_order α]
+
+@[simp] lemma succ_le_succ_of_le {a b : α} (h : a ≤ b) :
+  succ a ≤ succ b :=
```

Comment:
I'm very unfamiliar with the `monotonicity` tactic, but I believe tagging this `@[mono]` is a good idea.
```suggestion
@[mono, simp] lemma succ_mono {a b : α} (h : a ≤ b) :
  succ a ≤ succ b :=
```

Reply:
The `mono` label isn't recognized, but I don't know whether the right way to solve this is by importing `tactic.monotonicity`. If somebody could advise us, that would be great.